pilot/pkg/model/service.go

```diff
@@ -467,6 +467,10 @@ type ServiceAttributes struct {
 	// ExportTo defines the visibility of Service in
 	// a namespace when the namespace is imported.
 	ExportTo map[Visibility]bool
+	// Target defines a hostname that the current headless 'alias' service is pointing at.
+	// For example Kubernetes-specific equivalent of such service would be a service of type "ExternalName". This field
+	// would be populated with its "externalName" field value.
+	Target Hostname
```

Comment:
AliasHostnames []Hostname

Reply:
How about "ExternalName" - instead of inventing other names and use cases.

If we want to support multiple - we can use a slice, but I don't think any API or environment allows this right now.